src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -97,13 +97,11 @@ public CommandResult execute(Model model) throws CommandException {
         if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
             throw new CommandException(MESSAGE_DUPLICATE_PERSON);
         }
-        boolean isEditingClass = !editPersonDescriptor.getAClass().get().classDateTime.equals("");
 
-        ClassStorage.saveClass(editedPerson, index.getOneBased());
-        if (isEditingClass) {
+        if (!editPersonDescriptor.hasEmptyClass()) {
+            ClassStorage.saveClass(editedPerson, index.getOneBased());
```

Comment:
I am a bit confused about the usage of this line here. I interpreted this line to store a `Class` to the `editedPerson` if the `editedPerson` has a valid `Class`. But in this case, since `editPersonDescriptor` does not have any `Class`-relevant update, why do we need such a line here?

Reply:
> I am a bit confused about the usage of this line here. I interpreted this line to store a `Class` to the `editedPerson` if the `editedPerson` has a valid `Class`. But in this case, since `editPersonDescriptor` does not have any `Class`-relevant update, why do we need such a line here?

After going deep into the code (for around one hour), I understood that this design is to address the bug that when another field of person gets edited, the `Class` field will be automatically voided. I hence attempted to remove this line to clean up the code but failed as this design decision is also highly relevant to the testing in order to maintain consistency.

Reply:
I have also attempted to write test case of modifying non-Class field to replicate the past bug scenario and failed as ClassStorage is not initialized in testing environment.